### PR TITLE
Fix NAV_STATE enum values

### PR DESCRIPTION
**Pull request by fiam**
Since POSHOLD_2D entries were removed, we need to add 2 dummy
entries to keep the old values as they were, since they are logged
to blackbox.

### CODE REVIEW

**Review by digitalentity — COMMENTED** [review]
Each navigation fsm state consumes flash since it's an array entry. However, we might reuse these entries for i.e. @giacomo892's CRUISE mode ;)

**Comment by fiam** [comment]
@digitalentity Reusing them means that someone will need to update the blackbox related tools to deal with different meanings. At 80 bytes per entry, I think we can afford the wasted flash.

**Comment by stronnag** [comment]
Of historic note only; we've changed them in a non-backwards compatible way at least 3 times before without complaints. mwp's bb replay tool keeps a little list  ...

**Comment by digitalentity** [comment]
It would probably be better to add a field with permanent ID to fsm state entries and have this permanent ID logged to blackbox. This will be quite future-proof.

**Comment by fiam** [comment]
Just noticed that we also have a seemingly unused value, `NAV_STATE_LAUNCH_MOTOR_DELAY`. I think we should just add additional field with the state permanent ID and log that instead.

File: src/main/navigation/navigation_private.h [hunk 1]
```diff
@@ -176,41 +176,82 @@ typedef enum {
     NAV_FSM_EVENT_COUNT,
 } navigationFSMEvent_t;
 
+// This enum is used to keep values in blackbox logs stable, so we can
+// freely change navigationFSMState_t.
```
Comment (digitalentity) [inline src/main/navigation/navigation_private.h:180]: I don't like the name `PUBLIC_STATE`, same naming as in box modes seems more reasonable - `PERSISTENT_ID`

File: src/main/navigation/navigation_private.h [hunk 2]
```diff
@@ -243,6 +284,7 @@ typedef enum {
 } navigationFSMStateFlags_t;
 
 typedef struct {
+    navigationPublicState_t             publicState;
```
Comment (digitalentity) [inline src/main/navigation/navigation_private.h]: `persistentId` ?

File: src/main/navigation/navigation_private.h (hunk 1 above)
Comment (fiam) [reply src/main/navigation/navigation_private.h:180]: Roger that

**Review by digitalentity — APPROVED** [review]

**Pull request merged**